File: exercises/practice/sum-of-multiples/.meta/src/reference/java/SumOfMultiples.java
import java.util.Arrays;
import java.util.stream.IntStream;

public class SumOfMultiples {
    private final int sum;

    public SumOfMultiples(int limit, int[] set) {
        sum = calculateSum(limit, set);
    }

    public int getSum() {
        return sum;
    }

    private int calculateSum(int limit, int[] set) {
        return IntStream.range(1, limit)
            .filter(n -> Arrays.stream(set).filter(s -> s != 0).anyMatch(s -> n % s == 0))
            .sum();
    }

}

File: go/sum-of-multiples/sum_of_multiples.go
package summultiples

func SumMultiples(limit int, divisors ...int) int {
	sum := 0
	for i := 1; i < limit; i++ {
		for _, d := range divisors {
			if d == 0 {
				continue
			}
			if i%d == 0 {
				sum += i
				break
			}
		}
	}
	return sum
}
